lib/stream.js

```diff
@@ -44,39 +44,37 @@ Stream.Stream = Stream;
 // Internal utilities
 try {
   const types = require('util').types;
-  if (types && typeof types.isUint8Array === 'function') {
-    Stream._isUint8Array = types.isUint8Array;
+  if (types && typeof types.isArrayBufferView === 'function') {
+    Stream._isArrayBufferView = types.isArrayBufferView;
   } else {
     // This throws for Node < 4.2.0 because there's no util binding and
     // returns undefined for Node < 7.4.0.
-    Stream._isUint8Array = process.binding('util').isUint8Array;
+    Stream._isArrayBufferView = process.binding('util').isArrayBufferView;
   }
 } catch (e) {
 }
 
-if (!Stream._isUint8Array) {
-  Stream._isUint8Array = function _isUint8Array(obj) {
-    return Object.prototype.toString.call(obj) === '[object Uint8Array]';
-  };
+if (!Stream._isArrayBufferView) {
+  Stream._isArrayBufferView = ArrayBuffer.isView;
 }
 
 const version = process.version.substr(1).split('.');
 if (version[0] === 0 && version[1] < 12) {
-  Stream._uint8ArrayToBuffer = Buffer;
+  Stream._typedArrayToBuffer = Buffer;
 } else {
   try {
     const internalBuffer = require('internal/buffer');
-    Stream._uint8ArrayToBuffer = function _uint8ArrayToBuffer(chunk) {
+    Stream._typedArrayToBuffer = function _typedArrayToBuffer(chunk) {
       return new internalBuffer.FastBuffer(chunk.buffer,
                                            chunk.byteOffset,
                                            chunk.byteLength);
     };
   } catch (e) {
   }
 
-  if (!Stream._uint8ArrayToBuffer) {
-    Stream._uint8ArrayToBuffer = function _uint8ArrayToBuffer(chunk) {
-      return Buffer.prototype.slice.call(chunk);
+  if (!Stream._typedArrayToBuffer) {
+    Stream._typedArrayToBuffer = function _typedArrayToBuffer(chunk) {
+      return Buffer.from(chunk.buffer, chunk.byteOffset, chunk.byteLength);
     };
   }
 }
```

Comment:
I think we can clean up this whole file. As long as it exports the two functions we need, it's ok to not have all those checks. readable-stream@3 supports Node 6+ anyway.